pkg/scheduler/util/utils_test.go

```diff
@@ -122,55 +124,104 @@ func TestMoreImportantPod(t *testing.T) {
-			currentNominatedNodeName: "node1",
-			expectedPatchRequests:    1,
-			expectedPatchData:        `{"status":{"nominatedNodeName":null}}`,
+			name: "Should make patch requests to clear node name",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo2"},
+					Status:     v1.PodStatus{NominatedNodeName: "node2"},
+				},
+			},
+			expectedPatchRequests: 2,
+			expectedPatchData:     `{"status":{"nominatedNodeName":null}}`,
+		},
+		{
+			name:                  "Should not make patch request if nominated node is already cleared",
+			pods:                  []*v1.Pod{},
+			expectedPatchRequests: 0,
 		},
 		{
-			name:                     "Should not make patch request if nominated node is already cleared",
-			currentNominatedNodeName: "",
-			expectedPatchRequests:    0,
+			name: "Should not be patched if NominatedNodeName is empty",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: ""},
+				},
+			},
+			expectedPatchRequests: 0,
+		},
+		{
+			name: "Should return error if patch fails",
+			pods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "foo1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+				{
+					// In this test, the pod named "err1" returns an error in the patch.
+					ObjectMeta: metav1.ObjectMeta{Name: "err1"},
+					Status:     v1.PodStatus{NominatedNodeName: "node1"},
+				},
+			},
+			patchError:            statusErr,
+			expectedPatchErrors:   []error{statusErr},
+			expectedPatchRequests: 2,
+			expectedPatchData:     `{"status":{"nominatedNodeName":null}}`,
 		},
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			actualPatchRequests := 0
-			var actualPatchData string
+			var actualPatchData []string
 			cs := &clientsetfake.Clientset{}
 			cs.AddReactor("patch", "pods", func(action clienttesting.Action) (bool, runtime.Object, error) {
 				actualPatchRequests++
 				patch := action.(clienttesting.PatchAction)
-				actualPatchData = string(patch.GetPatch())
-				// For this test, we don't care about the result of the patched pod, just that we got the expected
-				// patch request, so just returning &v1.Pod{} here is OK because scheduler doesn't use the response.
-				return true, &v1.Pod{}, nil
+				// If the pod name is "err", return an error.
+				if patch.GetName() == "err1" {
+					return true, nil, test.patchError
+				} else {
+					actualPatchData = append(actualPatchData, string(patch.GetPatch()))
+					// For this test, we don't care about the result of the patched pod, just that we got the expected
+					// patch request, so just returning &v1.Pod{} here is OK because scheduler doesn't use the response.
+					return true, &v1.Pod{}, nil
+				}
```

Comment:
nits:

```suggestion
				} 
				actualPatchData = append(actualPatchData, string(patch.GetPatch()))
				// For this test, we don't care about the result of the patched pod, just that we got the expected
				// patch request, so just returning &v1.Pod{} here is OK because scheduler doesn't use the response.
				return true, &v1.Pod{}, nil
				
```